plugins/inputs/cisco_telemetry_mdt/cisco_telemetry_mdt.go

```diff
@@ -296,7 +371,9 @@ func (c *CiscoTelemetryMDT) handleTelemetry(data []byte) {
 		// Parse keys
 		tags = make(map[string]string, len(keys.Fields)+3)
 		tags["source"] = msg.GetNodeIdStr()
-		tags["subscription"] = msg.GetSubscriptionIdStr()
+		if msg.GetSubscriptionIdStr() != "" {
+			tags["subscription"] = msg.GetSubscriptionIdStr()
+		}
```

Comment:
We can save a function call here
```suggestion
		if msgID := msg.GetSubscriptionIdStr(); msgID != "" {
			tags["subscription"] = msgID
		}
```